lalrpop/src/lexer/dfa/overlap.rs

```diff
@@ -30,8 +30,8 @@ pub fn remove_overlap(ranges: &Set<Test>) -> Vec<Test> {
 
     let mut disjoint_ranges = vec![];
 
-    for &range in ranges {
-        add_range(range, 0, &mut disjoint_ranges);
+    for range in ranges {
+        add_range(range.clone(), 0, &mut disjoint_ranges);
```

Comment:
```suggestion
        add_range(range.to_owned(), 0, &mut disjoint_ranges);
```
No real difference, but to avoid to looking redundant clone. `to_owned()` will clear range is reference.